File: Post_Management.UI/src/app/core/models/api-response.ts
export interface ApiResponse<T> {
  status_code: number;
  message: string;
  reason?: string;
  is_success: boolean;
  data?: T;
}

export interface PaginationMeta {
  totalPages: number;
  currentPage: number;
  pageSize: number;
  totalItems: number;
}

export interface PagingResponse<T> {
  items: T[];
  meta: PaginationMeta;
} 

File: Post_Management.API/Data/Models/ApiResponse.cs
﻿using System.Text.Json.Serialization;

namespace Post_Management.API.Data.Models
{
    public class ApiResponse<T>
    {
        [JsonPropertyName("status_code")]
        public int StatusCode { get; set; }

        [JsonPropertyName("message")]
        public string Message { get; set; }

        [JsonPropertyName("reason")]
        public string? Reason { get; set; }

        [JsonPropertyName("is_success")]
        public bool IsSuccess { get; set; }

        [JsonPropertyName("data")]
        public T? Data { get; set; }
    }
}
